tools/manifest/fastjson.py

```diff
@@ -0,0 +1,162 @@
+import re
+import json
+
+from six import PY3
+
+
+MYPY = False
+if MYPY:
+    # MYPY is set to True when run under Mypy.
+    from typing import Any, AnyStr, Callable, Dict, IO, Text
+
+
+try:
+    import ujson
+except ImportError:
+    has_ujson = False
+else:
+    has_ujson = True
+
+try:
+    import orjson
+except ImportError:
+    has_orjson = False
+else:
+    has_orjson = True
+
+#
+# load
+#
+
+if has_orjson:
+    def load(fp):
+        # type: (IO[AnyStr]) -> Any
+        return orjson.loads(fp.read())
+
+elif has_ujson:
+    load = ujson.load
+
+else:
+    load = json.load
+
+
+#
+# loads
+#
+
+if has_orjson:
+    loads = orjson.loads  # type: Callable[[AnyStr], Any]
+
+elif has_ujson:
+    loads = ujson.loads
+
+else:
+    loads = json.loads
+
+
+#
+# dump/dumps_local options for some libraries
+#
+__ujson_dump_local_kwargs = {
+    'ensure_ascii': False,
+    'escape_forward_slashes': False,
+    'indent': 1,
+}  # type: Dict[str, Any]
+
+if PY3:
+    __ujson_dump_local_kwargs['reject_bytes'] = True
+
+
+__json_dump_local_kwargs = {
+    'ensure_ascii': False,
+    'indent': 1,
+    'separators': (',', ': '),
+}  # type: Dict[str, Any]
+
+
+#
+# dump_local (for local, non-distributed usage of JSON)
+#
+
+# orjson is disqualified here because orjson.dumps returns bytes
+
+if has_ujson:
+    def dump_local(obj, fp):
+        # type: (Any, IO[str]) -> None
+        return ujson.dump(obj, fp, **__ujson_dump_local_kwargs)
+
+else:
+    def dump_local(obj, fp):
+        # type: (Any, IO[str]) -> None
+        return json.dump(obj, fp, **__json_dump_local_kwargs)
+
+
+#
+# dumps_local (for local, non-distributed usage of JSON)
+#
+
+# orjson is disqualified here because orjson.dumps returns bytes
+
+if has_ujson:
+    def dumps_local(obj):
+        # type: (Any) -> Text
+        return ujson.dumps(obj, **__ujson_dump_local_kwargs)
+
+else:
+    def dumps_local(obj):
+        # type: (Any) -> Text
+        return json.dumps(obj, **__json_dump_local_kwargs)
+
+
+#
+# dump/dumps_dist (for distributed usage of JSON where files should safely roundtrip)
```

Comment:
I think I'd use "stable" rather than "dist". And maybe just make it a kwarg.

Reply:
"stable" to me implies merely that it's idempotent, rather than it produces the same result in all configurations (which is what we aim for for the MANIFEST as far as I'm aware).

Reply:
OK, well I don't really mind either way.